src/Provider/AzureMaps/AzureMaps.php

```diff
@@ -69,7 +69,7 @@ class AzureMaps extends AbstractHttpProvider implements Provider
      * AzureMaps constructor.
      */
     public function __construct(
-        HttpClient $client,
+        ClientInterface $client,
```

Comment:
Would you get a PHP error if you didn't change this?

Reply:
PHP says it's fine, but it will make this PR useless as you are still tied to PHP-HTTP as HTTP-client.